profile-widget/profilewidget2.cpp

```diff
@@ -644,8 +644,20 @@ void ProfileWidget2::plotDive(struct dive *d, bool force)
 
 #ifndef SUBSURFACE_MOBILE
 	if (plotInfo.maxhr) {
-		heartBeatAxis->setMinimum(plotInfo.minhr);
-		heartBeatAxis->setMaximum(plotInfo.maxhr);
+		int heartBeatAxisMin = lrint(plotInfo.minhr / 5.0 - 0.5) * 5;
+		int heartBeatAxisMax, heartBeatAxisTick;
+		if (plotInfo.maxhr - plotInfo.minhr < 40)
+			heartBeatAxisTick = 10;
+		else if (plotInfo.maxhr - plotInfo.minhr < 80)
+			heartBeatAxisTick = 20;
+		else if (plotInfo.maxhr - plotInfo.minhr < 100)
+			heartBeatAxisTick = 25;
+		else
+			heartBeatAxisTick = 50;
+		for (heartBeatAxisMax = 0; heartBeatAxisMax < plotInfo.maxhr; heartBeatAxisMax += heartBeatAxisTick);
```

Comment:
Something like
```
heartBeatAxisMax = ((plotInfo.maxhr - 1) / heartBeatAxisTick + 1) * heartBeatAxisTick ;
```
?

Reply:
Why not the for loop? :-) It's easy to understand.
But very good that you again pointed at this because I recognized that the for loop itself was incorrect. I put an updated version of the for loop.
But ok if we want to convert this into a calculation...